openshift-storage-libs/openshiftstoragelibs/baseclass.py

```diff
@@ -1097,3 +1098,33 @@ def verify_if_more_than_n_percentage_pod_restarted(
             msg = "Out of {} pods {} pods restarted {}".format(
                 len(pods_new), len(pods_restart), pods_restart)
             raise AssertionError(msg)
+
+    def _validate_gluster_pod_memory(self, size_limit):
+        """Validate memory usage for glusterfsd process"""
+
+        # Cmd to fetch the pid and memory usage
+        get_glusterds_pid = "pgrep glusterfsd"
+        get_mem_usage = "pmap -x {} | tail -1"
+
+        # Fetch gluster node/pod list
+        for gluster_node in self.gluster_servers:
+            # Fetch pid from the node/pod
+            out = cmd_run_on_gluster_pod_or_node(
+                self.ocp_master, get_glusterds_pid, gluster_node)
+            self.assertTrue(
+                out, "Failed to get pid from node/pod {}".format(gluster_node))
+            pid_list = out.split("\n")
+
+            # Fetch the memory usage for each pid
+            for pid in pid_list:
+                out = cmd_run_on_gluster_pod_or_node(
+                    self.ocp_master, get_mem_usage.format(pid), gluster_node)
+                self.assertTrue(
+                    out, "Failed to fetch the memory used from the node/pod "
```

Comment:
"Failed to fetch the memory used for glusterfsd process from the node/pod {}".format(gluster_node)

Reply:
Done